substrate/frame/identity/src/lib.rs

```diff
@@ -959,6 +973,100 @@ pub mod pallet {
+		/// - `target`: The account for which to reap identity state.
+		/// - `num_subs`: Witness data. The maximum number of sub accounts that `target` has.
+		///
+		/// Origin must be the `ReapOrigin`.
+		#[pallet::call_index(15)]
+		#[pallet::weight(T::WeightInfo::reap_identity(*num_subs))]
+		pub fn reap_identity(
+			origin: OriginFor<T>,
+			target: AccountIdLookupOf<T>,
+			num_subs: u32,
+		) -> DispatchResult {
+			// `ReapOrigin` to be set to `EnsureSigned<AccountId>` (i.e. anyone) on chain where we
+			// want to reap data (i.e. Relay) and `EnsureRoot` on chains where we want "normal"
+			// functionality (i.e. the People Chain).
+			T::ReapOrigin::ensure_origin(origin)?;
+			let who = T::Lookup::lookup(target)?;
+
+			// `take` any storage items keyed by `target`
+			let id = <IdentityOf<T>>::take(&who).ok_or(Error::<T>::NotNamed)?;
+			let fields = id.info.additional.len() as u32;
+			let (subs_deposit, sub_ids) = <SubsOf<T>>::take(&who);
+			// check witness data
+			let actual_subs = sub_ids.len() as u32;
+			ensure!(actual_subs <= num_subs, Error::<T>::TooManySubAccounts);
+			for sub in sub_ids.iter() {
+				<SuperOf<T>>::remove(sub);
+			}
+
+			// unreserve any deposits
+			let deposit = id.total_deposit().saturating_add(subs_deposit);
+			let err_amount = T::Currency::unreserve(&who, deposit);
+			debug_assert!(err_amount.is_zero());
+
+			// Finally, call the handler.
+			T::ReapIdentityHandler::on_reap_identity(&who, fields, actual_subs)?;
+			Self::deposit_event(Event::IdentityReaped { who });
+			Ok(())
+		}
+
+		/// Update the deposits held by `target` for its identity info.
+		///
+		/// Parameters:
+		/// - `target`: The account for which to update deposits.
+		///
+		/// May be called by any signed origin.
+		#[pallet::call_index(16)]
+		#[pallet::weight(T::WeightInfo::poke_deposit())]
+		pub fn poke_deposit(origin: OriginFor<T>, target: AccountIdLookupOf<T>) -> DispatchResult {
+			// anyone or root (so that the system can call it for identity migration)
+			let _ = ensure_signed_or_root(origin)?;
+			let target = T::Lookup::lookup(target)?;
+
+			// Identity Deposit
+			ensure!(IdentityOf::<T>::contains_key(&target), Error::<T>::NoIdentity);
+			let new_id_deposit = IdentityOf::<T>::try_mutate(
+				&target,
+				|registration| -> Result<BalanceOf<T>, DispatchError> {
+					if let Some(reg) = registration {
+						// Calculate what deposit should be
+						let field_deposit = BalanceOf::<T>::from(reg.info.additional.len() as u32)
+							.saturating_mul(T::FieldDeposit::get());
+						let new_id_deposit = T::BasicDeposit::get().saturating_add(field_deposit);
+
+						// Update account
+						let _ = Self::rejig_deposit(&target, reg.deposit, new_id_deposit)?;
+
+						reg.deposit = new_id_deposit;
+						return Ok(new_id_deposit)
+					} else {
+						return Err(Error::<T>::NoIdentity.into())
+					}
+				},
+			)?;
+
+			// Subs Deposit
+			let (current_subs_deposit, subs_of) = SubsOf::<T>::take(&target);
```

Comment:
Let's use `SubsOf::<T>::try_mutate` instead of `take` and `insert`. The code below just changes the value in the map, but the entry stays.